### PR TITLE
refactor account, general cleanup

### PR DESCRIPTION
**Pull request by benlemasurier**
account:
- split account related functions into separate files
- include all available json fields in account type

build:
- enable debugging symbols
- be pedantic

notes:
- I've used standard include guards rather than '#pragma once'. I'm open
  to change but I think this is a +1 for portability.

### CODE REVIEW

**Comment by benlemasurier** [comment]
I'm second guessing myself on the choice of `int account_get(struct account *a)`. What do you think, should we just let the function allocate the account struct and return `NULL` on error? This would make the management of arrays much easier for complex data types in the future.

File: src/sea.c [hunk 1]
```diff
@@ -2,97 +2,45 @@
...
-	droplet_limit = json_object_get(data, "droplet_limit");
-	if(!json_is_integer(droplet_limit))
-	{
-		fprintf(stderr, "error: droplet_limit is not a number\n");
-		ret = 1;
-		goto finish;
-	}
+	printf("Email: %s (verified: %s)\nStatus: %s\nStatus Message: %s\nUUID: %s\nDroplet Limit: %d\nFloating IP Limit: %d\n",
+			a->email,
+			(a->email) ? "true" : "false",
```
Comment (kitschysynq) [inline src/sea.c]: Should be `a->email_verified`

File: src/sea.h [hunk 2]
```diff
@@ -1,8 +1,7 @@
-#pragma once
-#include <stdio.h>
+#ifndef _SEA_H
```
Comment (kitschysynq) [inline src/sea.h:1]: Not of a fan of `#pragma` ? ^_^ We should probably lay out what OS/compiler combos (or C std) we want to support here.

[edit: didn't see your note, my bad!]

**Comment by kitschysynq** [comment]
This is much cleaner, thanks! I think that sig is good for the moment but we can certainly change it once we get into more realistic use-cases. For accounts, it's not even clear how we would manage an array of them at the moment, since we're assuming a single key for requests and managing that from within the process just seems... messy.

File: src/sea.c (hunk 1 above)
Comment (benlemasurier) [inline src/sea.c]: oops, nice catch

File: src/sea.h (hunk 2 above)
Comment (benlemasurier) [inline src/sea.h:1]: re: standards, c99? I think we should shoot for greatest portability, keep it simple, plain C. e.g., no `_GNU_SOURCE` extensions.

**Comment by benlemasurier** [comment]
> I think that sig is good for the moment but we can certainly change it once we get into more realistic use-cases. 

Sounds good, this will do for now. I started to question myself when poking at `/account/keys`. I'll rework this if the other way around is cleaner in the end.

File: src/sea.h (hunk 2 above)
Comment (kitschysynq) [inline src/sea.h:1]: Yeah, that would make sense, in general. FWIW `#pragma once` is supported across a fair number of very common compiler chains: https://en.wikipedia.org/wiki/Pragma_once#Portability

**Pull request merged**